consul/base.py

```diff
@@ -1838,3 +1839,210 @@ def peers(self):
             """
             return self.agent.http.get(
                 lambda x: json.loads(x.body), '/v1/status/peers')
+
+    class Query(object):
+        def __init__(self, agent):
+            self.agent = agent
+
+        def list(self, dc=None, token=None):
+            """
+            Lists all the active  queries. This is a privileged endpoint,
+            therefore you will only be able to get the prepared queries
+            which the token supplied has read privileges to.
+            """
+            params = {}
+            token = token or self.agent.token
+            if token:
+                params['token'] = token
+            if dc:
+                params['dc'] = dc
+
+            return self.agent.http.get(
+                lambda x: json.loads(x.body), '/v1/query', params=params)
+
+        def create(self, service,
```

Comment:
I'm pretty sure mashing the features of Consul's query into a flat function signature is going to come unstuck. For example, if more DNS options besides ttl are added, or if additional template types are added.

I think we'll want something similar to the Check conveniences to allow succinct expression of the full complexity of the query api. Not a huge issue immediately, except for backwards compatibility if we try and address this later. /cc @abn fyi..